### DISCUSSION TITLE
Not being able to make a request behind a proxy

### DISCUSSION THREAD

**Discussion by Elias-Yuri-Maximo** [microsoft/BotFramework-Composer#9689] [Q&A]
Hello, I am working behind a proxy, and I am trying to make an HTTP request using my bot. I have tested the endpoint and the parameters on postman, and It works fine. But when using the bot, It keeps giving me the following error request : 

"reason: getaddrinfo ENOTFOUND"

 I tried configuring proxies on NPM configs, environment variables and the request headers, but no success with that. 

My question is: When I am making the request in the bot, where does the proxy configuration is inherited from? does it come from the my OS variables, or NPM? Or is there a way to configure them through headers? I haven't found any similar thread on the internet yet...

**Comment by OEvgeny** [comment]
The `HTTPS_PROXY` environment variable affects only Composer requests and doesn't affect the requests made by bots. As well as NPM related environment variables affect only npm proxy settings.

For bot requests, the runtime running the bot is responsible for making the requests. To my knowledge neither .Net nor NodeJS don't have built-in ways of configuring http agents with proxy settings and don't respect system-level proxy settings out of the box.

I see a couple of ways to work around this issue:
- Modify bot source code to respect proxy settings. For example: for NodeJS there is a `global-agent` package which makes proxy configurable through environment variables
- Use a different method to access the restricted resources: e.g. a VPN connection would work as it moves the host resolution down to the network layer which is handled by OS

Helpful links:
- [Extend your Composer bot with code](https://learn.microsoft.com/en-us/composer/concept-extend-with-code)
- [global-agent npm package](https://github.com/gajus/global-agent)
